### PR TITLE
Added support for run arguments

### PR DESCRIPTION
**Pull request by ethauvin**
Added CLI option to include run arguments, e.g.:

```
./bld run --args="arg1 arg2 ..."
```

I couldn't figure out how to add a test for this. I tested manually, and it seems to work with everything I could think of.

### CODE REVIEW

File: src/main/java/rife/bld/operations/RunOperation.java [hunk 1]
```diff
@@ -52,6 +54,22 @@ public RunOperation fromProject(BaseProject project) {
         if (project.usesRife2Agent()) {
             operation.javaOptions().javaAgent(project.getRife2AgentFile());
         }
+
+        // parse the run arguments if any
+        var args = project.arguments();
```
Comment (gbevin) [inline src/main/java/rife/bld/operations/RunOperation.java:59]: Not sure about this, the intended use of the arguments is that each operation consumes the next arguments it needs, not any arguments anywhere passed to the command line. By only consuming the next ones, it's possible to freely chain commands on the command line without them interfering with each other

File: src/main/java/rife/bld/publish/PomBuilder.java [hunk 2]
```diff
@@ -121,6 +124,18 @@ public String build() {
         return StringUtils.stripBlankLines(t.getContent());
     }
 
+    /**
+     * Generates a POM into the given file.
+     *
+     * @since 1.7.1
+     */
+    public void generateInto(PublishInfo info, DependencyScopes dependencies, File file)
```
Comment (gbevin) [inline src/main/java/rife/bld/publish/PomBuilder.java:132]: I would either make this static and create a temporary `PomBuilder` instance for this method, or have `generateInto(File file)` to generate the current PomBuilder instance's output. I don't like that this `generateInto` has the side effect of changing the internal state.

File: src/main/java/rife/bld/operations/RunOperation.java (hunk 1 above)
Comment (ethauvin) [reply src/main/java/rife/bld/operations/RunOperation.java:59]: I'm not sure what you mean. It only deals with `./bld/run` arguments as far as my testing went.  e.g.;

`./bld co run -args="blah" blah publish`

will fail with bld complaining about blah being an invalid argument.

File: src/main/java/rife/bld/publish/PomBuilder.java (hunk 2 above)
Comment (ethauvin) [reply src/main/java/rife/bld/publish/PomBuilder.java]: Fixed

File: src/main/java/rife/bld/publish/PomBuilder.java [hunk 3]
```diff
@@ -121,6 +124,19 @@ public String build() {
...
     }
 
+    /**
+     * Generates a POM into the given file.
+     *
+     * @since 1.7.1
+     */
+    public static void generateInto(PublishInfo info, DependencyScopes dependencies, File file)
+            throws FileUtilsErrorException {
+        var pomBuilder = new PomBuilder();
```
Comment (gbevin) [inline src/main/java/rife/bld/publish/PomBuilder.java:134]: Maybe `var pom_builder = new PomBuilder().info(info).dependencies(dependencies);`

then remove the next two lines?

Comment (ethauvin) [reply src/main/java/rife/bld/publish/PomBuilder.java]: Back to my original thoughts. Then you are also creating a couple additional objects. Yet it matches the builder pattern used throughout the code base.

Done. :wink:

**Pull request merged**